ambry-account/src/test/java/com/github/ambry/account/HelixAccountServiceTest.java

```diff
@@ -332,23 +328,23 @@ public void receiveBadTopic() throws Exception {
    * Tests a number of bad inputs.
    */
   @Test
-  public void testNullInputs() {
+  public void testNullInputs() throws IOException {
     try {
-      new MockHelixAccountServiceFactory(null, new MetricRegistry(), notifier).getAccountService();
+      new MockHelixAccountServiceFactory(null, new MetricRegistry(), notifier, null).getAccountService();
       fail("should have thrown");
     } catch (NullPointerException e) {
       // expected
     }
 
     try {
-      new MockHelixAccountServiceFactory(vHelixConfigProps, null, notifier).getAccountService();
+      new MockHelixAccountServiceFactory(vHelixConfigProps, null, notifier, null).getAccountService();
       fail("should have thrown");
     } catch (NullPointerException e) {
       // expected
     }
-
-    new MockHelixAccountServiceFactory(vHelixConfigProps, new MetricRegistry(), null).getAccountService();
-
+    accountService =
+        new MockHelixAccountServiceFactory(vHelixConfigProps, new MetricRegistry(), null, null).getAccountService();
+    accountService.close();
     accountService = mockHelixAccountServiceFactory.getAccountService();
```

Comment:
what's happening here? There was a new factory used whose account service was closed - then an account service was obtained from the factory that is a class variable - what is the intent of the previous two lines?

Reply:
`accountService = mockHelixAccountServiceFactory.getAccountService();` is not needed. The purpose of the test is to test `null` parameters, and in this case it just ensures no exception is thrown if `notifier` is `null`.